File: frontend/Assets/Scripts/DataTransfer/Server/NewGameCmd.cs
using System.IO;
using Attributes;
using Core;
using DataTransfer.Data;


namespace DataTransfer.Server {

    [DTO (DTOCode.NewGame)]
    public class NewGameCmd : ServerCommand {

        public GameInitData Data { get; private set; }


        public override void ReadMembers (BinaryReader reader) {
            Data = new GameInitData ();
            Data.ReadMembers (reader);
        }


        public override void Execute () {
            The.SceneSwitcher.Load (Scenes.Battle, Data);
        }

    }

}

File: backend/src/main/java/dto/cmd/server/NewGameCmd.java
package dto.cmd.server;

import dto.DTOCode;
import dto.DTOs;
import io.netty.buffer.ByteBuf;


@DTOCode(DTOs.NEW_GAME)
public class NewGameCmd extends ServerCommand {

    @Override
    protected void writeMembers (ByteBuf byteBuf) {

    }


    @Override
    protected void readMembers (ByteBuf buffer) {

    }
}
